contracts/utils/IBCUUPSUpgradeable.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.28;
+
+import { IIBCUUPSUpgradeableErrors } from "../errors/IIBCUUPSUpgradeableErrors.sol";
+import { ContextUpgradeable } from "@openzeppelin-upgradeable/utils/ContextUpgradeable.sol";
+import { UUPSUpgradeable } from "@openzeppelin-contracts/proxy/utils/UUPSUpgradeable.sol";
+import { IIBCUUPSUpgradeable } from "../interfaces/IIBCUUPSUpgradeable.sol";
+
+/// @title IBC UUPSUpgradeable contract
+/// @notice This contract is an abstract contract for managing upgradability of IBC contracts.
+/// @dev This contract is developed with OpenZeppelin's UUPS upgradeable proxy pattern.
+/// @dev This contract is meant to be inherited by ICS26Router implementation, and it manages its own upgradability.
+/// @dev Other IBC contracts can directly query ICS26Router for the admin addresses to authorize UUPS upgrades (see
+/// ICS20Transfer).
+/// @dev This contract manages two roles: the multisig admin, and the governance admin. The multisig admin represents a
+/// timelocked
+/// security council, and the governance admin represents an interchain account from the governance of a counterparty
```

Comment:
Would it not make more sense if there was just a single admin that was the upgradeAdmin, which also has permission to swap itself out for govAdmin?

Reply:
This is a problem if govAdmin gets stuck due to light client freeze or another similar issue.

Reply:
Yes, agreed 👍